src/_pytest/deprecated.py

```diff
@@ -11,13 +11,45 @@
 
 from __future__ import annotations
 
+import sys
+from typing import TYPE_CHECKING
 from warnings import warn
 
 from _pytest.warning_types import PytestDeprecationWarning
 from _pytest.warning_types import PytestRemovedIn9Warning
 from _pytest.warning_types import UnformattedWarning
 
 
+# the `as` indicates explicit re-export to type checkers
+# mypy currently does not support overload+deprecated
+if sys.version_info >= (3, 13):
+    from warnings import deprecated as deprecated
+elif TYPE_CHECKING:
+    from typing_extensions import deprecated as deprecated
+else:
+
+    def deprecated(reason: str = "") -> object:
+        def decorator(func: object) -> object:
+            return func
+
+        return decorator
```

Comment:
[`warnings.deprecated`](https://docs.python.org/3/library/warnings.html#warnings.deprecated) will emit a warning at runtime in addition to be used by type checkers, but not for Python <= 3.12, which IIUC is a problem then.

Seems like the only solution is to always generate the runtime warning ourselves?

Reply:
hm, since this PR does manually raise the warning everywhere we should be getting double warnings on 3.13 ... and either the tests are not catching it or the second one is getting suppressed.